AutoGraph/WebSocketClient.swift

```diff
@@ -298,6 +325,41 @@ extension WebSocketClient: WebSocketDelegate {
         }
     }
     
+    func processWebSocketTextEvent(text: String) throws {
+        guard let data = text.data(using: .utf8) else {
+            throw ResponseSerializerError.failedToConvertTextToData(text)
+        }
+        
+        let json = try JSONValue.decode(data)
+        guard case let .string(typeValue) = json["type"] else {
+            return
+        }
+        
+        let graphQLWSProtocol = GraphQLWSProtocol(rawValue: typeValue)
+        switch graphQLWSProtocol {
+        case .connectionAck:
+            self.isConnectionAck = true
+            self.connectionAckWorkItem?.cancel()
+            self.connectionAckWorkItem = nil
+        case .data:
+            let subscriptionResponse = try self.subscriptionSerializer.serialize(text: text)
+            self.didReceive(subscriptionResponse: subscriptionResponse)
+        case .connectionKeepAlive:
+            self.writeQueuedSubscriptions()
+        case .error:
+            throw ResponseSerializerError.webSocketError(text)
```

Comment:
is this a different kind of error than a normal graphql error attached the subscription payload?

seems odd that there would be yet another error over this protocol since there's already websocket errors and gql errors

Reply:
it is a web socket error